### PR TITLE
Added cluster discovery mechanism

### PR DESCRIPTION
**Pull request by mikouaj**

### CODE REVIEW

File: internal/app/app.go [hunk 1]
```diff
@@ -236,12 +292,14 @@ func newConfigFromCli(cliConfig *CliConfig) *Config {
 	config := &Config{}
 	config.SilentMode = cliConfig.SilentMode
 	config.CredentialsFile = cliConfig.CredentialsFile
-	config.Clusters = []ConfigCluster{
-		{
-			Name:     cliConfig.ClusterName,
-			Location: cliConfig.ClusterLocation,
-			Project:  cliConfig.ProjectName,
-		},
+	if cliConfig.ClusterName != "" || cliConfig.ClusterLocation != "" || cliConfig.ProjectName != "" {
```
Comment (ewojtach) [inline internal/app/app.go:295]: shouldn't we require all three fields being filled in?

File: internal/app/config_test.go [hunk 2]
```diff
@@ -42,10 +46,18 @@ func TestReadConfig(t *testing.T) {
 		"- local: %s\n"+
 		"- repository: %s\n"+
 		"  branch: %s\n"+
-		"  directory: %s\n",
+		"  directory: %s\n"+
+		"clusterDiscovery:\n"+
```
Comment (ewojtach) [inline internal/app/config_test.go:50]: we can add also this options to user documentation

File: internal/app/app.go [hunk 3]
```diff
@@ -106,19 +110,15 @@ func (p *PolicyAutomationApp) ClusterReview() error {
 		return err
 	}
 
+	clusterIds, err := p.getClusters()
+	if err != nil {
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply internal/app/app.go:114]: Thanks for spotting that out, I forgot the `return` statement

File: internal/app/app.go [hunk 4]
```diff
@@ -89,6 +90,9 @@ func (p *PolicyAutomationApp) Close() error {
 	if p.gke != nil {
 		return p.gke.Close()
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply internal/app/app.go:91]: Good catch, we should close discovery client regardless on errors from gke client

File: internal/app/app_test.go [hunk 5]
```diff
@@ -23,9 +23,32 @@ import (
...
 	"gopkg.in/yaml.v2"
 
+	"github.com/google/gke-policy-automation/internal/gke"
 	"github.com/google/gke-policy-automation/internal/outputs"
 )
 
+type DiscoveryClientMock struct {
+	GetClustersInProjectFn func(name string) ([]string, error)
+	GetClustersInFolderFn  func(number string) ([]string, error)
+	GetClustersInOrgFn     func(number string) ([]string, error)
```
[the remark this answers is not in the document]
Comment (mikouaj) [reply internal/app/app_test.go:33]: Way of mocking without mock lib :-)

**Review by rickygodoy — APPROVED** [review]

**Pull request merged**